### ISSUE TITLE
Mac SDK include path apocalypse

### ISSUE THREAD

**Issue by jjwilke** [sstsimulator/sst-core#434]
Newer versions of xcode will no longer have a /usr/include in the path, meaning many macport/homebrew/default compilers will NOT bring basic library headers into the include path by default. The MAC SDK must be explicitly specified with `-isysroot`

We have a couple ways to fix this. One is to give an explicit
````
--with-sdk=PATH_TO_SDK
````
option and force the user to specify it. Otherwise, we can try to detect OS and version and find ourselves using
````
xcrun --sdk macosx --show-sdk-path
````

**Comment by berquist** [comment]
This doesn't seem to be an issue anymore.

**Comment by berquist** [comment]
Nope, I was wrong.  I am finding that, on macOS 15.5 (and possibly a little earlier on the 15.x line) with Apple Clang 17, with Command Line Tools installed but not XCode, the C++ compiler doesn't include the STL properly.
